src/xenia/apu/audio_decoder.cpp

```diff
@@ -0,0 +1,239 @@
+/**
+******************************************************************************
+* Xenia : Xbox 360 Emulator Research Project                                 *
+******************************************************************************
+* Copyright 2015 Ben Vanik. All rights reserved.                             *
+* Released under the BSD license - see LICENSE in the root for more details. *
+******************************************************************************
+*/
+
+#include "xenia/apu/audio_decoder.h"
+
+#include "xenia/apu/audio_system.h"
+#include "xenia/base/logging.h"
+
+extern "C" {
+  #include "libavcodec/avcodec.h"
+}
+
+// Credits for most of this code goes to:
+// https://github.com/koolkdev/libertyv/blob/master/libav_wrapper/xma2dec.c
+
+namespace xe {
+namespace apu {
+
+AudioDecoder::AudioDecoder() : offset_(0) {
```

Comment:
should also initialize the pointers to nullptr so that the ~dtor won't crash if Initialize() fails